.travis.yml

```diff
@@ -0,0 +1,63 @@
+language: rust
+
+env:
+  global:
+    - CRATE_NAME=easage
+    - FEATURES=clap
+    # 'rust:' statements in lint stage also need to be modified
+    - NIGHTLY=nightly-2018-01-01
+    - RUSTFMT=0.3.4
+    - CLIPPY=0.0.177
+
+stages:
+  - name: lint
+    if: tag IS blank
+  - name: build
+    if: tag IS blank
+  - name: deploy
+    if: tag IS present
+
+jobs:
+  fast_finish: true
+  include:
+    - stage: lint
+      rust: nightly-2018-01-01
+      env: rustfmt
+      before_install:
+        - if ! cargo +$NIGHTLY fmt --version | grep -q $RUSTFMT; then travis_wait cargo +$NIGHTLY install rustfmt-nightly --force --vers $RUSTFMT; fi
+      script:
+        - cargo +$NIGHTLY fmt -- --write-mode=diff
+    - stage: lint
+      rust: nightly-2018-01-01
+      env: clippy
+      before_install:
+        - if [[ `cargo +$NIGHTLY clippy -- --version` != $CLIPPY* ]] ; then travis_wait cargo +$NIGHTLY install clippy --force --vers $CLIPPY; fi
+      script:
+        - cargo +$NIGHTLY clippy --features "$FEATURES" -- -D warnings
+    - stage: build
+      script:
+        - cargo build --features "$FEATURES"
```

Comment:
FWIW `cargo test` (and `run`) will perform a `build` if cargo determines it is necessary.

Reply:
This is true, might switch to just using `cargo test` instead of `cargo build` completely, I wasn't sure if build does something extra or not though.

About `cargo run`, I found out something interesting while having `cargo run --feature clap easage` without any arguments. The [SubcommandRequiredElseHelp](https://github.com/kbknapp/clap-rs/blob/bc9ab21934cf657723633ee5b94e77f11a2884f7/src/app/parser.rs#L1128-L1135) enum you are using will cause `easage` to exit with a non-zero error code (i.e fail) when ran without arguments.
You can reproduce this with `easage ; echo $?`

This caused CI to fail if cargo run isn't ran with some argument that yields error level of `0` like `cargo run --feature clap -- --version`

Reply:
Indeed -- we should not be executing `cargo run` in a CI environment.